lib/api/request.js

```diff
@@ -1,14 +1,16 @@
+const last = require('ramda/src/last')
+const prop = require('ramda/src/prop')
+
 class Request {
 
   constructor(httpClient){
     this.httpClient = httpClient
   }
 
-  createHeaders(accessToken) {
-    const headers = {}
-    if (accessToken) headers.Authorization = `Bearer ${accessToken}`
-
-    return headers
+  createHeaders(callerMethodArgs) {
+    const lastArg = last(callerMethodArgs)
+    if (lastArg && prop('accessToken', lastArg)) return {Authorization: `Bearer ${lastArg.accessToken}`}
```

Comment:
Podle mě stačí použít klasické `lastArg.accessToken` namísto `prop()`. Jaká je výhoda? Jestli to chceš ale použít, tak můžeš napsat podmínku jinak:
```
if(prop('accessToken', lastArg)) ...
```

Aspoň podle tohodle editoru mi to vychází dobře i pro undefined lastArg. https://goo.gl/kUVx5K Nevýhodou bude, že čtenář tomu nebude moc rozumět, ale třeba to začneme používat. ;)

Reply:
Jo a zase ten ENTER za IF. ;)

Reply:
Zvlastni - me to bez checku prakticky nefunguje:
TypeError: Cannot read property 'accessToken' of undefined

Predtim to bylo psane jinak nez jsem sloucil ty podminky, mel jsem to jeste optimalizovat. :+1: